qubesbuilder/plugins/source/scripts/download-and-verify-file

```diff
@@ -0,0 +1,125 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+set -e
+[ "$DEBUG" = "1" ] && set -x
+
+usage() {
+echo "Usage: $(basename "$0") [OPTIONS]...
+This script downloads and verifies a file.
+
+Options:
+    --file-url         File URL
+    --checksum-file    Checksum for file to be downloaded
+    --checksum-method  Checksum method (default: sha256sum)
+    --signature-url    Signature file URL
+    --pubkey-file      GPG public key file to be used for verification
+    --output-dir       Output directory
+"
+}
+
+if ! OPTS=$(getopt -o hf:c:m:s:p:o: --long help,file-url:,checksum-file:,checksum-cmd:,signature-url:,pubkey-file:,output-dir: -n "$0" -- "$@"); then
+    echo "ERROR: Failed while parsing options."
+    exit 1
+fi
+
+eval set -- "$OPTS"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help) usage ;;
+        -f | --file-url ) FILE_URL="$2"; shift ;;
+        -c | --checksum-file ) FILE_CHECKSUM="$2"; shift ;;
+        -m | --checksum-cmd ) CHECKSUM_CMD="$2"; shift ;;
+        -s | --signature-url ) SIGNATURE_URL="$2"; shift ;;
+        -p | --pubkey-file ) PUBKEY_FILE="$2"; shift ;;
+        -o | --output-dir ) OUTPUT_DIR="$2"; shift ;;
+    esac
+    shift
+done
+
+if [ -z "${FILE_URL}" ]; then
+    printf "ERROR: Please provide file URL.\n"
+    exit 1
+fi
+FETCH_CMD=(curl --proto '=https' --proto-redir '=https' --tlsv1.2 --http1.1 -sSfL -o)
+FILE_NAME="$(basename "$FILE_URL")"
+
+if [ -z "${FILE_CHECKSUM}" ] && [ -z "${SIGNATURE_URL}" ] && [ -z "${PUBKEY_FILE}" ]; then
+    echo "ERROR: Please provide either CHECKSUM or SIGNATURE_URL and PUBKEY_FILE."
+    exit 1
+fi
+
+if [ -n "${FILE_CHECKSUM}" ] && ! [ -e "${FILE_CHECKSUM}" ]; then
+    echo "ERROR: Cannot find '${FILE_CHECKSUM}' checksum for '${FILE_NAME}'."
+    exit 1
+fi
+
+if [ -n "${SIGNATURE_URL}" ] && [ -z "${PUBKEY_FILE}" ]; then
+    echo "ERROR: Please provide PUBKEY_FILE to be used with SIGNATURE_URL."
+    exit 1
+fi
+
+if [ -z "${SIGNATURE_URL}" ] && [ -n "${PUBKEY_FILE}" ]; then
+    echo "ERROR: Please provide SIGNATURE_URL to be used with PUBKEY_FILE."
+    exit 1
+fi
+
+if [ -n "${SIGNATURE_URL}" ] && [ ! -e "${PUBKEY_FILE}" ]; then
+    echo "ERROR: Cannot find '${PUBKEY_FILE}' pubkey for '${SIGNATURE_URL}'."
+    exit 1
+fi
+
+# Download file with untrusted suffix
+"${FETCH_CMD[@]}" "${FILE_NAME}.untrusted" "$FILE_URL"
```

Comment:
Please use a temporary file (created by `mktemp`) for this, or (better yet) a Python or Rust program using `O_TMPFILE`.  `rpmcanon` does that.

Reply:
Do you mind to review the content of the scripts in `fetch` and possibly open PR. Thanks.